src/pages/index.js

```diff
@@ -0,0 +1,330 @@
+  }, 1);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+
+  if (outsideClickHandler && escapePressHandler) {
+    removeListeners(outsideClickHandler, escapePressHandler);
+  }
+
+  outsideClickHandler = null;
+  escapePressHandler = null;
+}
+
+function addListeners(outsideClickHandler, escapePressHandler) {
+  document.addEventListener("click", outsideClickHandler);
+  document.addEventListener("keydown", escapePressHandler);
+}
+
+function removeListeners(outsideClickHandler, escapePressHandler) {
+  document.removeEventListener("click", outsideClickHandler);
+  document.removeEventListener("keydown", escapePressHandler);
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+
+  const submitButton = evt.submitter;
+  submitButton.textContent = "Saving...";
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+    })
+    .catch(console.error)
+    .finally(() => {
+      submitButton.textContent = "Save";
+      disableButton(profileSubmitButton, settings);
+      closeModal(editProfileModal);
+    });
+}
+
+function handleCardFormSubmit(evt) {
+  evt.preventDefault();
+  const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
+
+  const submitButton = evt.submitter;
+  submitButton.textContent = "Saving...";
+
+  api
+    .addCard(inputValues)
+    .then((card) => {
+      const cardElement = getCardElement(card);
+      cardsList.prepend(cardElement);
+      evt.target.reset();
+    })
+    .catch(console.error)
+    .finally(() => {
+      submitButton.textContent = "Save";
+      disableButton(cardSubmitButton, settings);
+      closeModal(addCardModal);
+    });
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+
+  const submitButton = evt.submitter;
+  submitButton.textContent = "Saving...";
+  api
+    .editAvatarInfo(avatarInput.value)
+    .then((data) => {
+      profileAvatar.src = avatarInput.value;
+    })
+    .catch(console.error)
+    .finally(() => {
+      submitButton.textContent = "Save";
+      disableButton(avatarSubmitButton, settings);
+      closeModal(addAvatarModal);
+    });
+}
+
+profileEditButton.addEventListener("click", () => {
+  editModalNameInput.value = profileName.textContent;
+  editModalDescriptionInput.value = profileDescription.textContent;
+  resetValidation(
+    profileEditButton,
+    editFormElement,
+    [editModalNameInput, editModalDescriptionInput],
+    settings
+  );
+  openModal(editProfileModal);
+});
+editModalCloseButton.addEventListener("click", () => {
+  closeModal(editProfileModal);
+});
+editFormElement.addEventListener("submit", handleEditFormSubmit);
+
+profileAddButton.addEventListener("click", () => {
+  openModal(addCardModal);
+});
+
+cardModalCloseButton.addEventListener("click", () => {
```

Comment:
**Could be improved**

You can make a universal handler for any close buttons.

It will look something like this:

```js
// Find all close buttons
const closeButtons = document.querySelectorAll('.modal__close');

closeButtons.forEach((button) => {
  // Find the closest popup only once
  const popup = button.closest('.modal');
  // Set the listener
  button.addEventListener('click', () => closePopup(popup));
});



```

That’s why we make universal css classes for similar elements: here it’s `modal__close` for any close button.

So now you can add 100 modals: their close buttons will be handled automatically.


```js
.then(res => {
                if (res.ok) {
                    return res.json();
                }
                return Promise.reject(`Error ${res.status}`);
            })

```

The code of `ok` checking is repeated in **every** request of the project. To avoid it you need to make a special method `_checkResponse` that will check any response from the server if it’s ok or not.